src/server.cpp

```diff
@@ -9,6 +8,33 @@ using boost::system::error_code;
 
 namespace pls = std::placeholders;
 
+std::string packetTypeToString(int type)
+{
+    switch (type)
+    {
+        case ACCESS_REQUEST: return "ACCESS_REQUEST";
+        case ACCESS_ACCEPT: return "ACCESS_ACCEPT";
+        case ACCESS_REJECT: return "ACCESS_REJECT";
+        case ACCOUNTING_REQUEST: return "ACCOUNTING_REQUEST";
+        case ACCOUNTING_RESPONSE: return "ACCOUNTING_RESPONSE";
+        case ACCESS_CHALLENGE: return "ACCESS_CHALLENGE";
+        case STATUS_SERVER: return "STATUS_SERVER";
+        case STATUS_CLIENT: return "STATUS_CLIENT";
+    }
+    return "uncnown";
+}
+
+void printPacket(const Packet& p)
+{
+    std::cout << "Packet type: " << packetTypeToString(p.type()) << "\n";
+
+    std::cout << "ID: " << int(p.id()) << "\n";
```

Comment:
`int(something)` is a C-style cast, and it is considered bad. You have two options here:
```c++
std::cout << "ID: " << static_cast<unsigned>(p.id()) << "\n";
```
(`unsigned` because `uint8_t` is unsigned)

or
```c++
std::cout << "ID: " << std::to_string(p.id()) << "\n";
```
(`std::to_string` does correct conversion for numeric types).